File: Frontend/src/types/room.ts
export type RoomStatus = 'Aktif' | 'Bakımda' | 'Kapalı';

export interface Room {
  id: number;
  room_number: string;
  floor: number;
  room_type: string;
  capacity: string;
  features: string;
  description: string;
  service_notes: string;
  price_per_night: number;
  is_clean: boolean;
  is_occupied: boolean;
  maintenance_status: string;
  status: RoomStatus;
}

export interface RoomCreate {
  room_number: string;
  floor: number;
  room_type: string;
  capacity: string;
  features?: string;
  description?: string;
  service_notes?: string;
  price_per_night: number;
  status?: RoomStatus;
}

export interface RoomUpdate {
  room_number?: string;
  floor?: number;
  room_type?: string;
  capacity?: string;
  features?: string;
  description?: string;
  service_notes?: string;
  price_per_night?: number;
  status?: RoomStatus;
}

export interface RoomFilters {
  floor?: number;
  room_type?: string;
  is_occupied?: boolean;
  is_clean?: boolean;
  status?: RoomStatus;
}

export interface RoomStats {
  total: number;
  occupied: number;
  available: number;
  dirty: number;
  maintenance: number;
}

File: Backend/app/models/room.py
from sqlalchemy import Column, Integer, String, Boolean, Float, Text, Enum as SQLEnum
from app.database import Base
import enum

class RoomStatusEnum(enum.Enum):
    """Oda durumu enum"""
    ACTIVE = "Aktif"
    MAINTENANCE = "Bakımda"
    CLOSED = "Kapalı"

class Room(Base):
    __tablename__ = "rooms"

    id = Column(Integer, primary_key=True, index=True)
    room_number = Column(String, unique=True, index=True, nullable=False)
    floor = Column(Integer, nullable=False, default=1)  # Kat bilgisi
    room_type = Column(String, nullable=False)  # "Suite", "Standard", "Deluxe"
    capacity = Column(String, default="2 kişilik")
    features = Column(String, default="")
    description = Column(Text, default="")  # Oda açıklaması
    service_notes = Column(Text, default="")  # Servis notu (kişi notu)
    price_per_night = Column(Float, nullable=False)
    is_occupied = Column(Boolean, default=False)
    is_clean = Column(Boolean, default=True)
    maintenance_status = Column(String, default="Functional")
    status = Column(SQLEnum(RoomStatusEnum), default=RoomStatusEnum.ACTIVE)  # Oda durumu

